articles/models.js

```diff
@@ -12,30 +12,42 @@ exports.selectArticleById = (article_id) => {
     })
 }
 
-exports.selectArticles = (sort_by = 'created_at', order = 'desc') => {
+exports.selectArticles = (topic, sort_by = 'created_at', order = 'desc') => {
 
   const validSortBy = ['title', 'topic', 'author', 'body', 'created_at', 'votes']
   const validOrder = ['asc', 'desc']
   const queryValues = []
 
+
   if (sort_by && !validSortBy.includes(sort_by)) {
     return Promise.reject({ status: 400, message: 'invalid query'})
   }
   if (order && !validOrder.includes(order)) {
     return Promise.reject({ status: 400, message: 'invalid query'})
   }
 
-  const queryStr = `
+  let queryStr = `
     SELECT articles.article_id, title, topic, articles.author, articles.created_at, articles.votes, article_img_url, 
     COUNT(comments.article_id)::int AS comment_count
     FROM articles 
     LEFT JOIN comments ON articles.article_id = comments.article_id
+  `
+
+  if (topic) {
+    queryStr += `WHERE topic=$1`
+    queryValues.push(topic)
+  }
+
+  queryStr += `
     GROUP BY articles.article_id
-    ORDER BY ${sort_by} ${order}
-  ;`
+    ORDER BY ${sort_by} ${order};
+  `
 
-  return db.query(queryStr)
+  return db.query(queryStr, queryValues)
   .then(({ rows }) => {
+    if (rows.length === 0) {
```

Comment:
This might need a refactor if you consider topics that have no articles but are valid